### PR TITLE
[ACA-823] automate undo copy

### PR DESCRIPTION
**Pull request by mmuller88**
## PR Checklist
Please check if your PR fulfills the following requirements:

```
- [x] The commit message follows our guidelines: https://github.com/Alfresco/alfresco-content-app/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
```

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application / Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: ACA-823

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/suites/actions/copy.test.ts [hunk 1]
```diff
@@ -127,18 +132,18 @@ describe('Copy content', () => {
...
     await loginPage.loginWith(username);
     done();
   });
 
-  afterAll(async (done) => {
-    await apis.user.nodes.deleteNodeById(sourceId);
-    await apis.user.sites.deleteSite(siteName);
-    done();
-  });
+  // afterAll(async (done) => {
```
Comment (DenysVuika) [inline e2e/suites/actions/copy.test.ts]: Can you please remove this code if it is not needed anymore?

Comment (adinapitul) [reply e2e/suites/actions/copy.test.ts]: it's needed! :)
it's probably commented out cause this is still WIP I assume

File: e2e/suites/actions/copy.test.ts [hunk 2]
```diff
@@ -589,144 +288,368 @@ describe('Copy content', () => {
...
+    it('Undo copy of a folder when a folder with same name already exists on the destination - [C306941]', async () => {
+      await searchInput.searchFor(folder1);
+      await dataTable.waitForBody();
+
+      undoCopyOfAFolder();
+    });
+
   });
 
+  async function copyAFile(destination) {
```
Comment (adinapitul) [inline e2e/suites/actions/copy.test.ts]: can we pass also the item / items to copy as parameters?

File: e2e/suites/actions/copy.test.ts [hunk 3]
```diff
@@ -589,144 +288,368 @@ describe('Copy content', () => {
...
+    await dataTable.selectItem(file4);
+    await toolbar.clickMoreActionsCopy();
+    await copyDialog.selectLocation('Personal Files');
+    await copyDialog.selectDestination(destination);
+    await copyDialog.clickCopy();
+    const msg = await page.getSnackBarMessage();
+    expect(msg).toContain('Copied 1 item');
+    expect(msg).toContain('Undo');
+
+    await page.clickSnackBarAction();
```
Comment (adinapitul) [inline e2e/suites/actions/copy.test.ts]: try moving this line before the 2 expects from above. 
I saw some "stale element reference" errors, and suspect this is cause when it actually gets to this line the snackbar is no longer visible

File: e2e/suites/actions/copy.test.ts [hunk 4]
```diff
@@ -127,8 +137,8 @@ describe('Copy content', () => {
     await apis.user.nodes.createFolder(folderSiteFav, docLibId);
     await apis.user.nodes.createFolder(folderSiteSearch, docLibId);
 
-    await apis.user.shared.waitForApi({ expect: 4 });
-    await apis.user.favorites.waitForApi({ expect: 7 });
+    await apis.user.shared.waitForApi({ expect: 5 });
+    await apis.user.favorites.waitForApi({ expect: 9 });
```
Comment (adinapitul) [inline e2e/suites/actions/copy.test.ts]: you will have 10 favorite items, cause when you create a site it is also marked as favorite by default.
either change this expect to 10 or move it before the site creation

**Pull request merged**